File: app/cmd.rm/summarize.r
#syntax varlist
pandoc.header("Summary statistics", 3)

ret <- matrix(, nrow = length(cmd$varlist), ncol = 5)
colnames(ret) = c("Obs", "Mean" , "Std. Dev." , "Min" , "Max" )
rownames(ret) = cmd$varlist

i <-0
for (name in cmd$varlist) {
	i <- i + 1
	var <- star.data[,name]
	if (is.factor(var)) {
		var <- as.numeric((var))
	}
	ret[i,1] = sum(!is.na(var))
	ret[i,2] = mean(var,na.rm = TRUE)
	ret[i,3] = sd(var,na.rm = TRUE)
	ret[i,4] = min(var,na.rm = TRUE)
	ret[i,5] = max(var,na.rm = TRUE)
}


pander(ret)

File: app/static/commands/summarize.js
// always available for directly accessing the command line, even without parsing anything
console.log(currentCommandLine);


input = parseStataSyntaxFromCommandLine({parseType:"varlist"});
var theVars = "'" + input.vars.join("','") + "'";

runPackage({
	title: "Summary statistics",
	input: input,
	rPackages: ["stargazer","knitr","summarytools"],
	rCommands: [
		"subset <- data.frame(data.matrix(stardata[,c("+theVars+")]));", // subsets, and removes labels
		"ret <- summarytools::descr(subset,stats=c('n.valid','mean','sd','min','max'),transpose=TRUE);",
		"ret$Mean <- round(ret$Mean, digits=3); ret$Std.Dev <- round(ret$Std.Dev, digits=3);",
		"cat('<style>.summarytable {width:100%} .summarytable td {padding:2px}</style>');" +
		"cat(knitr::knit_print(knitr::kable(ret,format='html', table.attr = 'class=\"summarytable\"')))"
	]
});